submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepositoryTest.kt

```diff
@@ -0,0 +1,117 @@
+package ac.uk.ebi.biostd.persistence.doc.db.data
+
+import ac.uk.ebi.biostd.persistence.doc.db.repositories.SubmissionMongoRepository
+import ac.uk.ebi.biostd.persistence.doc.integration.MongoDbReposConfig
+import ac.uk.ebi.biostd.persistence.doc.model.DocProcessingStatus
+import ac.uk.ebi.biostd.persistence.doc.model.DocProcessingStatus.PROCESSED
+import ac.uk.ebi.biostd.persistence.doc.model.DocProcessingStatus.PROCESSING
+import ac.uk.ebi.biostd.persistence.doc.model.DocSection
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmission
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionMethod
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+import java.time.Instant
+
+@ExtendWith(SpringExtension::class)
+@Testcontainers
+@SpringBootTest(classes = [MongoDbReposConfig::class])
+internal class SubmissionDocDataRepositoryTest {
+
+    @Autowired
+    lateinit var testInstance: SubmissionDocDataRepository
+
+    @Autowired
+    lateinit var submissionMongoRepository: SubmissionMongoRepository
+
+    @Nested
+    inner class UpdateStatus {
+        @Test
+        fun `successful status update`() {
+            submissionMongoRepository.save(testDocSubmission("accNo1", 1, PROCESSING))
+
+            testInstance.updateStatus(PROCESSED, "accNo1", 1)
```

Comment:
I think using the named `version` parameter would add more clarity to the tests

Reply:
this is really sugetive one may same the same about status and processes status not sure what convention of logic conclusion we can arrive to add parameters names or not